test/extended/networking/services.go

```diff
@@ -65,4 +69,93 @@ var _ = Describe("[sig-network] services", func() {
 			Expect(checkServiceConnectivity(f1, f2, DIFFERENT_NODE)).To(Succeed())
 		})
 	})
+
+	InBareMetalClusterContext(oc, func() {
+		f := e2e.NewDefaultFramework("external-ip")
+		It("ensures external ip policy and auto assign cidr is configured correctly on the cluster", func() {
+			// Test a load balancer service with default cluster networks config
+			// In this case service must not assigned with ingress ip address.
+			By("create service of type load balancer with default cluster networks config")
+			serviceName := "test-service-without-external-ip"
+			createPodForService(f, serviceName, "")
+			createWebserverLBService(f, serviceName, "", []string{})
+			By("check load balance service not having ingress ip configured")
+			Eventually(func() string {
+				service, err := f.ClientSet.CoreV1().Services(f.Namespace.Name).Get(context.Background(), serviceName, metav1.GetOptions{})
+				expectNoError(err)
+				if len(service.Status.LoadBalancer.Ingress) == 0 {
+					return ""
+				}
+				return service.Status.LoadBalancer.Ingress[0].IP
+			}, 5*time.Minute, 1*time.Second).Should(Equal(""))
+
+			// Test external ip policy configured with allowedCIDRs. Make sure service
+			// is assigned with an ip address that is with allowedCIDRs range
+			By("update network config with allowed cidr for external ip")
```

Comment:
Can you first ensure that creating a service with ExternalIP fails? That is the most critical part of this test.

Reply:
yes, done.